indolaw-parser/parser_utils.py

```diff
@@ -318,56 +318,156 @@ def clean_law(law: List[str]) -> List[str]:
     law = [line.strip() for line in law]
     law = [' '.join(line.split()) for line in law]
 
-    save_law_to_file(law, 'a.txt')
-
     '''
-    Remove semantically meaningless text e.g '. . .' or '1 / 23'
-
-    Mostly, they come in whole lines, but sometimes they get squashed
-    onto the end of real lines
-
-    e.g a real line ending in '2 / 43' in UU 18 2017
+    Preps for cleaning stages for potential save in a dict of stages
+    TODO(Mel):  1. Maybe change law_stages headers into enums?
+                2. Extend saving mechanism to catch and save during a crash
+                3. Add documentation on how saving mechanism works 
     '''
-    law = list(filterfalse(ignore_line, law))
-    law = clean_squashed_page_numbers(law)
+    law_stages = [['clean_squashed_page_numbers',[]], ['clean_maybe_list_items',[]],
```

Comment:
Can we change this data structure to something like
```
law_stages = [
      { 
             'stage': CleaningStage.CLEAN_SQUASHED_PAGE_NUMBERS,
             'law': [],
      },
     ...
]
```

In general, pick the data structure that makes the logic easiest to read, especially for 
- a) other people, and 
- b) you 5 months from now when you've forgotten how this PR was written 😛 

W/ the current data structure, the code is littered with stuff like `law_stages[1][1]` where I have no idea what the second array index means at all. If we had something like `law_stages[1]['law']` then that would at least give more of an idea. 

For example, now when I read `len(law_stages[int_user_input_stage - 1][1]) == 0` I know that the len = 0 check means you're checking if the law has any lines i.e we already cleaned that stage.

I think the TLDR here is dictionaries are your friend haha (I'm writing a slightly longer comment here since I noticed you did this too in the first parser PR you made)

Reply:
Yeah I agree! I actually started this using dict for the same reason you mentioned above, but I changed it into a list for _some_ reason I can't remember now... Probably because I forgot OrderedDict existed (I wanted to preserve the ordering of the stages, but if I use enums as stage headings, would it matter?).

Reply:
Ignore my previous comment (for multiple reasons). Will implement as you've stated 👍